### DISCUSSION TITLE
Best design practice for form that creates data in multiple tables

### DISCUSSION THREAD

**Discussion by studiopatagon** [jeremyevans/roda#317] [General]
Just wondering if anyone has any advice on the best design practice for something along these lines:

Form is posted with article title, content, photo, etc that go into the articles table.  But also with 'categories' and 'photos' that have their own table.

1. Is the best way to handle it from the route so that it starts by creating and saving a new 'article' model, then if successful it does the same with categories and photos?

OR

2. Is it best to have the article model handle the rest ?

I've always done #1 with my CMS/PHP work. 

Thanks for any recommendations!

**Comment by adam12** [comment]
I'm assuming you're using Sequel?

You have a few options (and probably more):

Use the [`nested_attributes` Sequel plugin](http://sequel.jeremyevans.net/rdoc-plugins/classes/Sequel/Plugins/NestedAttributes.html) which will create the associated records for you if you follow the expected form param format.

Open a transaction and create everything separately. You will have to handle your own validation errors here tho.

```ruby
DB.transaction do
  article = Article.create(r.params["article"]) # Consider using .new and #set_fields instead
  photo = Photo.create(r.params["photo"])
  article.add_photo(photo)
  # etc
end
```

**Reply by studiopatagon** [reply]
Yes, Sequel. Thanks, I will look at that plugin. I was about to go the DB.transaction way...
